src/typing.py

```diff
@@ -1221,9 +1221,9 @@ def __getitem__(self, parameters):
         elif args == []:
             parameters = ((), result)
         else:
-            if not isinstance(args, list):
-                raise TypeError("Callable[args, result]: args must be a list."
-                                " Got %.100r." % (args,))
+            if not isinstance(args, list) or ... in args or () in args:
```

Comment:
Hm... It really shouldn't be necessary to check this here. I think the reason may be that you're trying to make the hash key for e.g. `Callable[[int, str], float]` be the tuple `(int, str, float)` rather than just nesting the original structure. The exceptions in `__getitem_inner__` seem due to this.

Or maybe you're trying to make `__parameters__` a single-level tuple? But there are two exceptions for this, one for `Callable[..., t]` and one for `Callable[[], t]` (IIUC). I don't think this is really worth it...